prost-build/src/code_generator.rs

```diff
@@ -1191,21 +1191,27 @@ fn unescape_c_escape_string(s: &str) -> Vec<u8> {
 ///
 /// It also tries to handle cases where the stripped name would be
 /// invalid - for example, if it were to begin with a number.
+///
+/// The stripped name can be `Self`, which is sanitized analogously to [to_upper_camel]
 fn strip_enum_prefix(prefix: &str, name: &str) -> String {
     let stripped = name.strip_prefix(prefix).unwrap_or(name);
 
     // If the next character after the stripped prefix is not
     // uppercase, then it means that we didn't have a true prefix -
     // for example, "Foo" should not be stripped from "Foobar".
-    if stripped
+    match if stripped
```

Comment:
I think this is not very readable. I suggest to first assigning the stripped value to a variable and then replace the keywords

Reply:
Will do